src/wallet/api/swapService.ts

```diff
@@ -172,3 +173,70 @@ export const get = async (proposalId: string, password: string): Promise<AtomicS
     throw err;
   }
 };
+
+interface SwapUpdateParams {
+  proposalId: string;
+  password: string;
+  partialTx: string;
+  version: number;
+  signatures?: string;
+}
+
+/**
+ * Updates the proposal on the Atomic Swap Service with the parameters informed
+ */
+export const update = async (params: SwapUpdateParams):
+  Promise<{ success: boolean }> => {
+  validateParameters();
+  const { proposalId, password, partialTx, version, signatures } = params;
+
+  const swapAxios = await axiosInstance();
+  const options = {
+    headers: { 'X-Auth-Password': hashPassword(password) }
+  } as AxiosRequestConfig;
+
+  const payload = {
+    partialTx: encryptString(partialTx, password),
+    version,
+    signatures: signatures ? encryptString(signatures, password) : null,
+  };
+
+  const { data } = await swapAxios.put<{ success: boolean }>(`/${proposalId}`, payload, options);
+  return { success: data?.success };
+
+  /**
+   * Validates the many mandatory parameters for the `update` method
+   * @throws {Error} if any mandatory parameter is missing
+   * @throws {Error} if any version parameter is invalid
+   */
+  function validateParameters() {
```

Comment:
💬 Interesting!

I'm concerned about readability. This structure caused me a little confusion while I was reading. Because of this, I believe it would be better to extract this validation function and do some refactoring to return an array of validation messages, like:

```javascript
function validateUpdateParameters(params) {
  const missingParameters = [];
  if (!params) {
    return { success: false, validationMsg: 'The positional function argument "params" is required' };
  }
  if (!proposalId) {
    missingParameters.push('proposalId');
  }
  // hidden for brevity...
  return { success: true, validationMsg: 'Missing mandatory parameters: ' + missingParameters.join(', ') + '.' };
}
```

This way you avoid throwing exceptions directly from the validation and pass this responsibility to the `update` function, turning the expectations explicit.

The `update` function turns out to be more easy to read and comprehend.

Reply:
It's a good pattern, but would diverge from the other `swapService` methods. All of them _throw_ on invalid parameters and _resolve_ with `{ success: boolean }` when receiving data from the server.

It could be a possibility for a future refactor, but I think it would not be a good idea on this PR.
What do you think?